### DISCUSSION TITLE
Optimization: use Franek-Jennings-Smyth (FJS) instead of Knuth-Morris-Pratt (KMP)

### DISCUSSION THREAD

**Discussion by dumblob** [vlang/v#15441] [Ideas]
Maybe *V* can become one of the leading languages when it comes to speed of substring search with [FJS](https://github.com/CGJennings/fjs-string-matching ).

**Comment by dumblob** [comment]
@penguindark FJS might be of interest for you also for other reasons as you seem to be dealing with strings quite frequently :wink:.

**Comment by penguindark** [comment]
I know it, may be in future we will implement it. tnx.

**Comment by dumblob** [comment]
Another algorithm to investigate would be the combination of "Aho-Corasick with Double-Array Trie" (Java implementation: https://github.com/hankcs/AhoCorasickDoubleArrayTrie ).

This one actually improves over Aho-Corasick so significantly that it makes the algorithm linear :open_mouth:. The only question is how expensive is the building phase in practice and only measurements can tell us.

**Comment by penguindark** [comment]
@dumblob very interesting,I'll take a look at it, Tnx :)
